cpg-analysis/src/main/kotlin/de/fraunhofer/aisec/cpg/query/QueryTree.kt

```diff
@@ -336,6 +345,34 @@ fun QueryTree<*>.successfulLastNodes(): List<Node> {
     return finallyTheEntirePaths.mapNotNull { (it as? List<*>)?.last() }.filterIsInstance<Node>()
 }
 
+/**
+ * @param assumptionType The type of assumption used to differentiate between assumptions and group
+ *   similar assumptions.
+ * @param aID an ID chosen by the caller to identify the assumption across translation, e.g.
+ *   filename or classname, with a function name and a potential counter.
+ * @param scope The scope that the assumption has validity for, here the scope is a node, because
+ *   the assumption is valid for every node in its ast subtree.
+ * @param message The message describing the assumption that was taken.
+ */
+fun <T> QueryTree<T>.assume(
```

Comment:
Please move all of the assume functions into one file

Reply:
That should not work if want to have this type of syntax, in that case i would not be allowed to use querytree.assume(...) but would have to write it down like this every time: `queryTree.also {it.assumptions.add(assume(...))}`. This is because the core module wants to have assumptions, when they are used in passes, but the core module has no dependency to cpg-analysis.

Reply:
What I would recommend is to have a common interface like "Assumable" (bad wording) and have the extension function based on that. You can then even have Node and Edge derive from this interface and reduce the set of functions from 4 to possibly 1 or 2.

Reply:
I made an interface `HasAssumptions` out of it with only one assume function